compiler/cli_reporter.nim

```diff
@@ -3398,12 +3442,17 @@ proc reportHook*(conf: ConfigRef, r: Report): TErrorHandling =
   # comment
   assertKind r
 
-  if conf.isEnabled(r) and r.category == repDebug and tryhack:
-    # Force write of the report messages using regular stdout if tryhack is
-    # enabled
+  if (conf.isEnabled(r) and r.category == repDebug and tryhack) or
+     # Force write of the report messages using regular stdout if tryhack is
+     # enabled
+     r.kind in rintCliKinds
+     # or if we are writing command-line help/usage information - it must
+     # always be printed
+    :
```

Comment:
Not really a big fan of this dangling `:`, maybe you can group the entire expression in `()` to increase clarity? Like this

```nim
if (
  longExpr1 or
  anotherLongExpr
):
  doStuff
```